server/src/main/java/org/cloudfoundry/identity/uaa/codestore/CodeStoreSecurityConfiguration.java

```diff
@@ -0,0 +1,52 @@
+package org.cloudfoundry.identity.uaa.codestore;
+
+import static org.cloudfoundry.identity.uaa.web.AuthorizationManagersUtils.anyOf;
+
+import org.cloudfoundry.identity.uaa.oauth.provider.authentication.OAuth2AuthenticationProcessingFilter;
+import org.cloudfoundry.identity.uaa.oauth.provider.error.OAuth2AccessDeniedHandler;
+import org.cloudfoundry.identity.uaa.oauth.provider.error.OAuth2AuthenticationEntryPoint;
+import org.cloudfoundry.identity.uaa.web.FilterChainOrder;
+import org.cloudfoundry.identity.uaa.web.UaaFilterChain;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.annotation.Order;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AnonymousConfigurer;
+import org.springframework.security.config.annotation.web.configurers.CsrfConfigurer;
+import org.springframework.security.web.authentication.preauth.AbstractPreAuthenticatedProcessingFilter;
+
+@Configuration
+@EnableWebSecurity
+class CodeStoreSecurityConfiguration {
+
+    @Bean
+    @Order(FilterChainOrder.CODESTORE)
+    UaaFilterChain codestore(
+            HttpSecurity http,
+            @Qualifier("resourceAgnosticAuthenticationFilter") OAuth2AuthenticationProcessingFilter oauth2ResourceFilter
+    ) throws Exception {
+        var originalFilterChain = http
+                .securityMatcher("/Codes/**")
```

Comment:
I've noticed a difference between XML and Java, where in XML

```
"/Codes/**"
```
matches both `/Codes` and `/Codes/`

I vaguely remember that `/Codes` would not match in Java.

I'm RTO, so limited equipment for a deep dive, but should this be:

```
.securityMatcher("/Codes", "/Codes/**")
```

asking for a friend!

Reply:
Good point. I just tested by running the UAA, and `/Codes/**` actually matches requests to `/Codes`, `/Codes/` and `/Codes/foobar`.

Validated with a simple test:

```java
var request = MockMvcRequestBuilders.get("/Codes").buildRequest(new MockServletContext())
new MvcRequestMatcher(new HandlerMappingIntrospector(), "/Codes/**")
  .matches(request); // -> returns true
```

Reply:
Found it!

The changes you are mentioning are introduced for Spring Framework 6.x, and are about trailing slashes ([spring-framework docs](https://docs.spring.io/spring-framework/reference/web/webmvc/filters.html#filters.url-handler)).

Reply:
Good find!